book/internal-services.md

```diff
@@ -0,0 +1,81 @@
+|-----------------|-------------------------------------------------------------------------------------------------------------------|-----------------------------------------------|
+| PipelineRun     | Once a InternalRequest has been reconciled and the Pipeline to run has been discovered and is ready for execution | To perform the steps in the Internal Pipeline |
+
+### READ
+
+| Custom Resource  | When?                                                       | Why?                                |
+|------------------|-------------------------------------------------------------|-------------------------------------|
+| Config           | During controller startup and during each reconcile attempt | To obtain configuration information |
+
+### UPDATE
+
+| Custom Resource  | When?                                                              | Why?                                                                      |
+|------------------|--------------------------------------------------------------------|---------------------------------------------------------------------------|
+| InternalRequest  | During the lifecycle of an attempt to complete an InternalRequest  | To provide status for the execution of the Pipeline to the remote cluster |
+
+### WATCH
+
+| Custom Resource  | When?                            | Why?                                                                               |
+|------------------|----------------------------------|------------------------------------------------------------------------------------|
+| InternalRequest  | Always                           | To provide an API to process an internal request                                   |
+| PipelineRun      | Once the PipelineRun is created  | To relay the Internal PipelineRun status to the remote InternalRequest for viewing |
+
+## Security Risk Mitigations
+
+Enabling remote clusters the ability to run internal jobs carrying certain security risks:
+
+* A user or a pipeline may attempt to run an internal job that it is not permitted to.
+* A user or a pipeline may attempt to run an arbitrary job
+
+The following list describes measures in places to mitigate those risks.
+
+* The creation of an `InternalRequest` custom resource requires permission on the Remote Cluster.
+* The Internal Services controller instance is configured to watch a specific cluster.
+  * This cluster is provided to the controller as an argument to a secret that was added by the admin team.
+  * The secret contains a KUBECONFIG file.
+* Only `Pipelines` that are defined and exist within the controller's namespace can be executed on the internal, private cluster.
+* The Internal Services controller only watches and acts on remote namespaces that are specifically allowed in the `Config` custom resource.
+
+## Detailed Workflow
```

Comment:
Is there a way that we can easily indicate where the actions are happening -- whether on the internal or remote clusters? I almost feel like swimlanes or a table would be better here.

Reply:
the system diagram shows this in detail. Is that diagram missing anything?

Reply:
The diagram is helpful when looking at it. The issue is likely the view I have where the diagram is separated more. Do the numbers here correspond to the numbers in the diagram? If so, indicating that would help.

Reply:
updated